docs/rfcs/0001-resource_routing.md

```diff
@@ -0,0 +1,111 @@
+                          PUT    /contacts/:contact_id/phone_numbers/:id(.:format)           phone_numbers#update
+                          DELETE /contacts/:contact_id/phone_numbers/:id(.:format)           phone_numbers#destroy
+                 contacts GET    /contacts(.:format)                                         contacts#index
+                          POST   /contacts(.:format)                                         contacts#create
+              new_contact GET    /contacts/new(.:format)                                     contacts#new
+             edit_contact GET    /contacts/:id/edit(.:format)                                contacts#edit
+                  contact GET    /contacts/:id(.:format)                                     contacts#show
+                          PATCH  /contacts/:id(.:format)                                     contacts#update
+                          PUT    /contacts/:id(.:format)                                     contacts#update
+                          DELETE /contacts/:id(.:format)                                     contacts#destroy
+    phone_number_contacts GET    /phone_numbers/:phone_number_id/contacts(.:format)          contacts#index
+                          POST   /phone_numbers/:phone_number_id/contacts(.:format)          contacts#create
+ new_phone_number_contact GET    /phone_numbers/:phone_number_id/contacts/new(.:format)      contacts#new
+edit_phone_number_contact GET    /phone_numbers/:phone_number_id/contacts/:id/edit(.:format) contacts#edit
+     phone_number_contact GET    /phone_numbers/:phone_number_id/contacts/:id(.:format)      contacts#show
+                          PATCH  /phone_numbers/:phone_number_id/contacts/:id(.:format)      contacts#update
+                          PUT    /phone_numbers/:phone_number_id/contacts/:id(.:format)      contacts#update
+                          DELETE /phone_numbers/:phone_number_id/contacts/:id(.:format)      contacts#destroy
+            phone_numbers GET    /phone_numbers(.:format)                                    phone_numbers#index
+                          POST   /phone_numbers(.:format)                                    phone_numbers#create
+         new_phone_number GET    /phone_numbers/new(.:format)                                phone_numbers#new
+        edit_phone_number GET    /phone_numbers/:id/edit(.:format)                           phone_numbers#edit
+             phone_number GET    /phone_numbers/:id(.:format)                                phone_numbers#show
+                          PATCH  /phone_numbers/:id(.:format)                                phone_numbers#update
+                          PUT    /phone_numbers/:id(.:format)                                phone_numbers#update
+                          DELETE /phone_numbers/:id(.:format)                                phone_numbers#destroy
+```
+
+The JSON API resource routing declaration would be:
+
+```ruby
+contraint mime_type: :json_api do
```

Comment:
`constraint` is for the wrong keyword here. What I want is more of a `concern` that extends `resources`.